controllers/storagecluster/reconcile.go

```diff
@@ -163,6 +164,11 @@ func (r *StorageClusterReconciler) Reconcile(ctx context.Context, request reconc
 		return reconcile.Result{}, err
 	}
 
+	// Set default mon count value in the StorageCluster CR
+	if sc.Spec.ManagedResources.CephCluster.MonCount == 0 {
+		sc.Spec.ManagedResources.CephCluster.MonCount = defaults.DefaultMonCount
```

Comment:
@umangachapagain @iamniting AFAIK we from the operator code should not set any fields in the CR Spec from the code, isn't this breaking that? Isn't just setting it from the API not enough?

Reply:
https://github.com/red-hat-storage/ocs-operator/blob/49da9ae0b3a85da5f141a865520f87995249be90/controllers/storagecluster/cephcluster.go#L717-L726
This will create a confusing state in arbiter mode, where the MonCount will be set to 3 on the storageCluster Cr, but for arbiter mode, we will pass the monCount as 5 to the CephCluster CR. So 3 on StorageCluster but 5 mons in actual and in the CephCluster will be confusing.

Reply:
Makes sense, but based on the use case probably it will be better to have the arbiter check in the updated code as well. WDYT?

Reply:
Mon count should be read from StorageCluster CR and then written to CephCluster CR.
Validations can be done to ensure valid Mon count is provided.

By default StorageCluster will not have any Mon count. So, internally, we decide to use `ArbiterModeMonCount` or `DefaultMonCount`. But, this is updated in CephCluster and not StorageCluster.

If StorageCluster Mon count is provided, and
1. if we have arbiter mode enabled, then use `ArbiterModeMonCount`.
2. if we have enough zones, then use 5.
3. if none of the above, use 3.

Technically, this shouldn't even be a user option as user does not have much choice here.

Reply:
@travisn What do you think of this?

- If we allow users to choose and say they choose MonCount: 3 and enable Arbiter mode, they will see 5 Mons.
- If they choose MonCount: 5 and they don't have 5 zones, they'd have multiple MONs in same zone which isn't desired.
- If they choose MonCount: 3 and disable arbiter but have 5 zones, we expect them to have 5 MONs, which isn't what they desired.

So, it looks like we are throwing a bunch of issues at the user to force them to use a MonCount that we want them to use.

If we still want to make it a user option, we should only allow them to set it to 5 or keep it undefined so we can apply our default.

Reply:
This use case of applying monCount was introduced to tackle the situation where a. user has 5 or more zones and they want to have 5 mons.
Right now, we have an alert configured that would tell them to update the mon count if they have 5 or more zones, but the user can choose to silence that alert. And, continue with 3 mons (if they want to).
This option should be only configured by an user if they have 5 or more zones in their cluster, otherwise it should be controlled by the operator and the appropriate value should be set, be it overriding the user's value.

We have validations in place that doesn't allow the user to set anything else apart from 3,5 for monCount field in the CR.

Reply:
If they desire 5 mons, where is it validated that they have 5 zones?

Reply:
We don't validate it at ocs-operator level as of now, it's just the `CephMonLowNumber` alert that will be shown to the user on the UI, if they have 5 or more zones.

Reply:
Yeah, that was the plan of the epic for the 4.15 release.